### DISCUSSION TITLE
Failed to connect to Web3Auth using node-sdk on Firebase cloud functions

### DISCUSSION THREAD

**Discussion by wegoinggroup** [Web3Auth/Web3Auth#1361] [Help]
I try to use the `@web3auth/node-sdk@1.0.1` to connect to Web3Auth with Firebase, but there's some errors.

>  {"severity":"ERROR","message":"Unhandled error Error: Error occurred while verifying paramscould not verify identity A7m6knG8vwY4besTiMPigSTn7Ew2 sub\n    at /Users/user/Workspace/wegoing/creatorc-functions/functions/node_modules/@toruslabs/torus.js/dist/torusUtils-node.js:491:20\n    at <anonymous>\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"}

---
Here is the sample code in the client web,
```
const idToken = await firebase.auth.currentUser?.getIdTokenResult(true);
const signinWallet = httpsCallable(functions, "signinWallet");
const result = await signinWallet({
  idToken: idToken?.token
})
```

and the code on cloud functions,
```
const web3auth = new Web3Auth({
    clientId: clientId,
    chainConfig: {
      chainNamespace: "eip155",
      chainId: "0x5",
      rpcTarget: "https://goerli.infura.io/v3/9aa3d95b3bc440fa88ea12eaa4456161",
    },
    web3AuthNetwork: "testnet",
  });
  await web3auth.init();
  const provider = await web3auth.connect({
    verifier: "firebase-verifier",
    verifierId: "sub",
    idToken: idToken,
  });
```

I also confirmed the `idToken` contains `sub`, `aud`, `iat`, `iss` field, etc.

Any idea why this error happened?

**Comment by wegoinggroup** [comment]
Finally notice the post [Web3Auth's Node.js SDK](https://github.com/orgs/Web3Auth/discussions/835) contained the solution.
The `sub` should be parsed from the `idToken` and filled into the `verifierId` field for the `web3auth.connect` function.

**Reply by itsdiay** [reply]
Wow thanks. Ran into the same issue, and this solved it for me. The docs don't say this at all
